include/stronk/stronk.h

```diff
@@ -87,6 +109,32 @@ concept stronk_like = requires(T v) {
     } -> std::convertible_to<typename T::underlying_type>;
 };
 
+template<typename StronkT>
+struct can_be_underlying_type
+{
+    template<typename T>
+    constexpr auto as() const -> const T&
+    {
+        auto& self = static_cast<const StronkT&>(*this);
+        if constexpr (std::same_as<T, typename StronkT::underlying_type>) {
+            return self.template unwrap<StronkT>();
+        } else {
+            return self.template as<T>();
```

Comment:
How does this work? Isn't it infinite recursion? (Same for non const overload)

Reply:
ahh I thought I had removed it - Ill delete